File: Structural - Exercises/Solutions-Leonie/Bridge/providers/outlook.py
from providers.provider_interface import ProviderInterface

class OutlookProvider(ProviderInterface):
    def authenticate(self):
        print("Authenticating with Microsoft credentials")

    def process_email(self, email_data):
        print("Processing email through Microsoft infrastructure")

    def logout(self):
        print("Outlook: Logging out from session")

File: Lesson.03 - Design Patterns - Structural/Bridge/Outlook.java
public class Outlook implements EmailProvider {
    @Override
    public void authenticate() {
        System.out.println("Outlook: Authenticating with Exchange credentials");
    }

    @Override
    public void processEmail(String emailData) {
        System.out.println("Outlook: Processing email through Microsoft Exchange");
    }

    @Override
    public void logout() {
        System.out.println("Outlook: Logging out from session");
    }
}
